ras_backstage/controllers/secure_messaging_controller.py

```diff
@@ -43,6 +43,27 @@ def get_message(encoded_jwt, message_id, is_draft):
     return json.loads(response.text)
 
 
+def get_thread_by_id(encoded_jwt, thread_id):
+    """
+    The method return all the message part of a thread.
+    :param encoded_jwt:  is the security token.
+    :param thread_id: is the id of the thread selected.
+    :return: a list of messages sent/received part of conversation thread
+    """
+
+    url = f"{app.config['RAS_SECURE_MESSAGING_SERVICE_THREAD']}{thread_id}"
+
+    headers = _create_authorization_header(encoded_jwt)
+    response = request_handler('GET', url, headers=headers)
+
+    if response.status_code != 200:
```

Comment:
Do we need different actions based on error code ranges?

Reply:
can be 200 or 500.  If you se eall teh other roots hav ethe same http status check.

Reply:
Agree, they all do this and we should have a review of all error handling in backstage but for now I think this should stay as it is for consistency

Reply:
changed